mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/PromoteDialog.kt

```diff
@@ -0,0 +1,138 @@
+package ch.epfl.sdp.mobile.ui.game
+
+import androidx.compose.animation.animateColorAsState
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.Icon
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.OutlinedButton
+import androidx.compose.material.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog as NativeDialog
+import androidx.compose.ui.window.DialogProperties
+import ch.epfl.sdp.mobile.state.LocalLocalizedStrings
+import ch.epfl.sdp.mobile.ui.PawniesColors
+import ch.epfl.sdp.mobile.ui.game.ChessBoardState.Rank.*
+import ch.epfl.sdp.mobile.ui.prepare_game.Dialog
+
+/**
+ * An icon which can be selected and indicates the choice of the user when it comes to promoting a
+ * piece.
+ *
+ * @param color the [ChessBoardState.Color] of the user who will promote a piece.
+ * @param rank the [ChessBoardState.Rank] of the promotion choice.
+ * @param selected true if the piece choice is currently selected.
+ * @param onClick a callback which will be called when the icon is pressed.
+ * @param modifier the [Modifier] for this composable
+ */
+@Composable
+private fun PromoteIcon(
+    color: ChessBoardState.Color,
+    rank: ChessBoardState.Rank,
+    selected: Boolean,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+  val strings = LocalLocalizedStrings.current
+  val tint by animateColorAsState(if (selected) PawniesColors.Orange200 else PawniesColors.Green800)
+  Icon(
+      painter = rank.icon(color = color),
+      contentDescription = rank.contentDescription(strings),
+      tint = tint,
+      modifier =
+          modifier
+              .clip(CircleShape)
```

Comment:
Why you use clip here ?

Reply:
To clip the ripple from the `clickable` modifier.